### PR TITLE
Add "Apply" button to entry and group edit windows

### PR DESCRIPTION
**Pull request by droidmonkey**
## Description

Add's an "Apply" buttons to both the entry and group edit windows. Implements #509.

## Motivation and context

Often times you want to save your changes without exiting the edit windows.

## How has this been tested?

Manually and with updated GUI tests.

## Types of changes

- ✅ New feature (non-breaking change which adds functionality)

## Checklist:

- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**
- ✅ I have compiled and verified my code with `-DWITH_ASAN=ON`. **[REQUIRED]**
- ✅ I have added tests to cover my changes.

### CODE REVIEW

File: src/gui/EditWidget.cpp [hunk 1]
```diff
@@ -102,7 +103,10 @@ void EditWidget::setReadOnly(bool readOnly)
         m_ui->buttonBox->setStandardButtons(QDialogButtonBox::Close);
     }
     else {
-        m_ui->buttonBox->setStandardButtons(QDialogButtonBox::Ok | QDialogButtonBox::Cancel);
+        m_ui->buttonBox->setStandardButtons(QDialogButtonBox::Ok | QDialogButtonBox::Cancel | QDialogButtonBox::Apply);
```
Comment (louib) [inline src/gui/EditWidget.cpp]: @droidmonkey maybe nitpicking, but I'd place the `Apply` before the `Cancel`.

File: src/gui/EditWidget.cpp [hunk 2]
```diff
@@ -102,7 +103,10 @@ void EditWidget::setReadOnly(bool readOnly)
...
+        // Find and connect the apply button
+        QPushButton *applyButton = m_ui->buttonBox->button(QDialogButtonBox::Apply);
```
Comment (louib) [inline src/gui/EditWidget.cpp]: @droidmonkey `QPushButton* applyButton`

File: src/gui/EditWidget.ui [hunk 3]
```diff
@@ -66,7 +66,7 @@
      <item>
       <widget class="QDialogButtonBox" name="buttonBox">
        <property name="standardButtons">
-        <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
+        <set>QDialogButtonBox::Apply|QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
```
Comment (louib) [inline src/gui/EditWidget.ui]: @droidmonkey we could also preserve the same order for the `Ok` `Apply` and `Cancel` buttons here.

File: src/gui/EditWidget.cpp (hunk 1 above)
Comment (weslly) [inline src/gui/EditWidget.cpp]: @louib It seems Qt ignores the order of the values and sets it according to the platform defaults:
https://doc.qt.io/qt-5/qmessagebox.html#the-property-based-api

**Comment by louib** [comment]
@droidmonkey the new button works great! However, it appears the button stays pushed until a click is made somewhere in the window. Would it be possible to never leave it pushed?

screenshot from 2017-06-11 21-30-58

Notice also that the Ok button is always pushed. I tested and it's the same thing on the current develop branch. Do you recall if there was a reason for that?
screenshot from 2017-06-11 21-30-41

**Comment by droidmonkey** [comment]
@louib the QDialogButtonBox handles all the particulars when it comes to button behavior and ordering. I am not aware of anything special we are doing that would cause the always pushed look to occur.

**Pull request merged**